### PR TITLE
Allow running testharness/testdriver/reftests in servodriver

### LINKED ISSUE #34523: Can't run WPT test suite with servodriver
**Describe the bug:**
Some WPT tests require testdriver.js, but it seems servodriver isn't working.
There are errors when we called its python scripts.

**To Reproduce:**

Choose a test requries testdervier.js and run it with `--product servodriver`.

```
./mach test-wpt --product servodriver tests/wpt/tests/pointerevents/pointerevent_click_during_capture.html 
```

And you should see similar error like this:

```
No build type specified so assuming `--dev`.
Running WPT tests with /servo/target/debug/servo
 0:01.07 wptserve INFO Create socket on: (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('127.0.0.1', 8889))
 0:01.08 wptserve INFO Bind on: (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('127.0.0.1', 8889))
 0:01.08 wptserve INFO Listen on: (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('127.0.0.1', 8889))
 0:01.08 wptserve INFO Starting https server on https://127.0.0.1:8445
 0:01.09 wptserve INFO Starting https server on https://127.0.0.1:8444
 0:01.10 wptserve INFO Starting http2 server on https://127.0.0.1:9000
 0:01.15 wptserve INFO Starting http server on http://127.0.0.1:8002
 0:01.16 wptserve INFO Starting https server on https://127.0.0.1:8446
 0:01.17 wptserve INFO Starting http server on http://127.0.0.1:8000
 0:01.17 wptserve INFO Create socket on: (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('127.0.0.1', 8888))
 0:01.17 wptserve INFO Bind on: (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('127.0.0.1', 8888))
 0:01.17 wptserve INFO Listen on: (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('127.0.0.1', 8888))
 0:01.18 wptserve INFO Starting http server on http://127.0.0.1:8001
 0:01.19 wptserve INFO Starting http server on http://127.0.0.1:8003
 0:01.19 wptserve INFO Starting https server on https://127.0.0.1:8443
 0:01.41 SUITE_START: web-platform-test - running 2 tests
 0:01.41 INFO Using 2 child processes
 0:01.41 ERROR Failure during init:
Traceback (most recent call last):
  File "/home/wusyong/Desktop/servo/tests/wpt/tests/tools/wptrunner/wptrunner/testrunner.py", line 261, in init
    self.browser.start(group_metadata=group_metadata, **self.browser_settings)
  File "/home/wusyong/Desktop/servo/tests/wpt/tests/tools/wptrunner/wptrunner/browsers/servodriver.py", line 142, in start
    processOutputLine=[self.on_output],
                       ^^^^^^^^^^^^^^
AttributeError: 'ServoWebDriverBrowser' object has no attribute 'on_output'

 0:01.41 ERROR Failure during init:
Traceback (most recent call last):
  File "/home/wusyong/Desktop/servo/tests/wpt/tests/tools/wptrunner/wptrunner/testrunner.py", line 261, in init
    self.browser.start(group_metadata=group_metadata, **self.browser_settings)
  File "/home/wusyong/Desktop/servo/tests/wpt/tests/tools/wptrunner/wptrunner/browsers/servodriver.py", line 142, in start
    processOutputLine=[self.on_output],
                       ^^^^^^^^^^^^^^
AttributeError: 'ServoWebDriverBrowser' object has no attribute 'on_output'

 0:01.41 CRITICAL Uncaught exception in TestRunnerManager.run:
Traceback (most recent call last):
  File "/home/wusyong/Desktop/servo/tests/wpt/tests/tools/wptrunner/wptrunner/testrunner.py", line 450, in run_loop
    new_state = f()
                ^^^
  File "/home/wusyong/Desktop/servo/tests/wpt/tests/tools/wptrunner/wptrunner/testrunner.py", line 618, in init
    return self.init_failed()
           ^^^^^^^^^^^^^^^^^^
  File "/home/wusyong/Desktop/servo/tests/wpt/tests/tools/wptrunner/wptrunner/testrunner.py", line 671, in init_failed
    self.stop_runner(force=True)
  File "/home/wusyong/Desktop/servo/tests/wpt/tests/tools/wptrunner/wptrunner/testrunner.py", line 961, in stop_runner
    self.browser.stop(force=True)
  File "/home/wusyong/Desktop/servo/tests/wpt/tests/tools/wptrunner/wptrunner/testrunner.py", line 290, in stop
    self.browser.stop(force=force)
  File "/home/wusyong/Desktop/servo/tests/wpt/tests/tools/wptrunner/wptrunner/browsers/servodriver.py", line 162, in stop
    self.output_handler.after_process_stop()
  File "/home/wusyong/Desktop/servo/tests/wpt/tests/tools/wptrunner/wptrunner/browsers/base.py", line 276, in after_process_stop
    self.start()
  File "/home/wusyong/Desktop/servo/tests/wpt/tests/tools/wptrunner/wptrunner/browsers/base.py", line 262, in start
    assert self.state == OutputHandlerState.AFTER_PROCESS_START
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError

 0:01.42 CRITICAL Uncaught exception in TestRunnerManager.run:
Traceback (most recent call last):
  File "/home/wusyong/Desktop/servo/tests/wpt/tests/tools/wptrunner/wptrunner/testrunner.py", line 450, in run_loop
    new_state = f()
                ^^^
  File "/home/wusyong/Desktop/servo/tests/wpt/tests/tools/wptrunner/wptrunner/testrunner.py", line 618, in init
    return self.init_failed()
           ^^^^^^^^^^^^^^^^^^
  File "/home/wusyong/Desktop/servo/tests/wpt/tests/tools/wptrunner/wptrunner/testrunner.py", line 671, in init_failed
    self.stop_runner(force=True)
  File "/home/wusyong/Desktop/servo/tests/wpt/tests/tools/wptrunner/wptrunner/testrunner.py", line 961, in stop_runner
    self.browser.stop(force=True)
  File "/home/wusyong/Desktop/servo/tests/wpt/tests/tools/wptrunner/wptrunner/testrunner.py", line 290, in stop
    self.browser.stop(force=force)
  File "/home/wusyong/Desktop/servo/tests/wpt/tests/tools/wptrunner/wptrunner/browsers/servodriver.py", line 162, in stop
    self.output_handler.after_process_stop()
  File "/home/wusyong/Desktop/servo/tests/wpt/tests/tools/wptrunner/wptrunner/browsers/base.py", line 276, in after_process_stop
    self.start()
  File "/home/wusyong/Desktop/servo/tests/wpt/tests/tools/wptrunner/wptrunner/browsers/base.py", line 262, in start
    assert self.state == OutputHandlerState.AFTER_PROCESS_START
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError
```

**Platform:**
All platforms are affected.

### PR DESCRIPTION
**Pull request by jdm**
These changes remove all of the custom integration code for the servodriver WPT executor, making it a tiny wrapper around the upstream webdriver executors, which should improve maintainability going forward. This PR also implements support for missing webdriver commands that are sent by the webdriver test executor, and extends the JS->webdriver serialization to cover arbitrary JS objects which is required by the integration with testharnessreport.js.

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #34523
- [x] These changes do not require tests because we don't use this test harness in CI yet. They were tested manually against tests/wpt/mozilla/tests and tests/wpt/tests/cookie with `--product servodriver`.

### CODE REVIEW

**Comment by wusyong** [comment]
@jdm Thanks for the help! I'm really glad tests require testdriver.js can run again.
Do you think it's possible merge servo and servodriver into one? Because @mrobinson mentioned it's better we can run WPT workflow with the same executor.

**Comment by mrobinson** [comment]
> @jdm Thanks for the help! I'm really glad tests require testdriver.js can run again. Do you think it's possible merge servo and servodriver into one? Because @mrobinson mentioned it's better we can run WPT workflow with the same executor in yesterday's triage meeting.

It's true that I think it'd be better to have one harness, but I'm happy if we move toward that situation gradually as the webdriver version of the harness improves.

File: components/shared/script/lib.rs [hunk 1]
```diff
@@ -808,6 +808,15 @@ pub enum WebDriverCommandMsg {
...
     ),
+    /// Create a new tab that loads about:blank. The constellation will use
+    /// the provided channels to return the top level browsing cntext id
+    /// associated with the new tab, and a notification when the initial
+    /// load is complete.
+    NewTab(IpcSender<TopLevelBrowsingContextId>, IpcSender<LoadStatus>),
+    /// Close the webview associated with the provided id.
+    CloseWindow(TopLevelBrowsingContextId),
+    /// Focus the webview associated with the provided id.
+    FocusWebView(TopLevelBrowsingContextId),
```
Comment (wusyong) [inline components/shared/script/lib.rs:819]: I feel like these variants should share the same term here. Could they all be `WebView` like [`ConstellationMsg`](https://github.com/servo/servo/blob/bc2def0582ff374d72633f2a06b47842483bac47/components/shared/compositing/constellation_msg.rs#L58) does?

**Comment by jdm** [comment]
> @jdm Thanks for the help! I'm really glad tests require testdriver.js can run again. Do you think it's possible merge servo and servodriver into one? Because @mrobinson mentioned it's better we can run WPT workflow with the same executor in yesterday's triage meeting.

I do think it's a realistic goal. Right now it would mean a regression in developer ergonomics, since the --debugger flag does not launch a debugger with the servodriver harness.

File: components/shared/script/lib.rs (hunk 1 above)
Comment (jdm) [reply components/shared/script/lib.rs]: Done.

**Comment by jdm** [comment]
```
error: patch failed: tools/wptrunner/wptrunner/browsers/servodriver.py:9
error: tools/wptrunner/wptrunner/browsers/servodriver.py: patch does not apply
error: patch failed: tools/wptrunner/wptrunner/executors/executorservodriver.py:5
error: tools/wptrunner/wptrunner/executors/executorservodriver.py: patch does not apply
error: patch failed: tools/wptrunner/wptrunner/testharnessreport-servodriver.js:1
error: tools/wptrunner/wptrunner/testharnessreport-servodriver.js: patch does not apply
```
Not a good sign.

**Review by wusyong — APPROVED** [review]

**Pull request merged**